### PR TITLE
KAFKA-17175: Remove interface `BrokerNode` and `ControllerNode`

### PR DESCRIPTION
**Pull request by frankvicky**
After [KAFKA-16560](https://issues.apache.org/jira/browse/KAFKA-16560), `BrokerNode` and `ControllerNode` are almost the same. Hence, it is time to remove them.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/java/kafka/testkit/TestKitNode.java [hunk 1]
```diff
@@ -28,7 +42,140 @@ default String metadataDirectory() {
...
+                private final MetaPropertiesEnsemble ensemble = copier.copy();
+
+                @Override
+                public MetaPropertiesEnsemble initialMetaPropertiesEnsemble() {
+                    return ensemble;
+                }
+
+                @Override
+                public Map<String, String> propertyOverrides() {
+                    return propertyOverrides;
```
Comment (chia7712) [inline core/src/test/java/kafka/testkit/TestKitNode.java]: please get a copy of `propertyOverrides` to make it immutable

File: core/src/test/java/kafka/testkit/TestKitNode.java [hunk 2]
```diff
@@ -28,7 +42,140 @@ default String metadataDirectory() {
...
```
Comment (chia7712) [inline core/src/test/java/kafka/testkit/TestKitNode.java]: please get a copy of `propertyOverrides` to make it immutable

File: core/src/test/java/kafka/testkit/TestKitNode.java [hunk 3]
```diff
@@ -28,7 +42,140 @@ default String metadataDirectory() {
...
+        return initialMetaPropertiesEnsemble().logDirProps().keySet();
+    }
+
     MetaPropertiesEnsemble initialMetaPropertiesEnsemble();
 
-    boolean combined();
+    Map<String, String> propertyOverrides();
+
+
+    class Builder {
```
Comment (chia7712) [inline core/src/test/java/kafka/testkit/TestKitNode.java]: Do we really need this builder? It seems `TestKitNodes#Builder` is almost same to this builder. Could we move helper method `buildBrokerNode` and `buildControllerNode` to  `TestKitNodes`?

**Review by chia7712 — COMMENTED** [review]
@frankvicky thanks for this nice refactor. a major comment is left. PTAL

**Comment by frankvicky** [comment]
Hi @chia7712 
I have remove redundant `Builder`, PTAL 😄

File: core/src/test/java/kafka/testkit/TestKitNodes.java [hunk 4]
```diff
@@ -216,4 +218,90 @@ public ListenerName externalListenerName() {
     public ListenerName controllerListenerName() {
         return new ListenerName("CONTROLLER");
     }
-}
+
+    public static TestKitNode buildBrokerNode(int id,
```
Comment (chia7712) [inline core/src/test/java/kafka/testkit/TestKitNodes.java]: `private static`

File: core/src/test/java/kafka/testkit/TestKitNodes.java [hunk 5]
```diff
@@ -216,4 +218,90 @@ public ListenerName externalListenerName() {
...
+            }
+
+            @Override
+            public Map<String, String> propertyOverrides() {
+                return Collections.unmodifiableMap(propertyOverrides);
+            }
+        };
+    }
+
+    public static TestKitNode buildControllerNode(int id,
```
Comment (chia7712) [inline core/src/test/java/kafka/testkit/TestKitNodes.java]: `private static`

**Review by chia7712 — COMMENTED** [review]
@frankvicky thanks for updates

**Comment by frankvicky** [comment]
Hello @chia7712 
I have redefine the scope of method, PTAL 😺

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**